src/System.Runtime/tests/System/ArrayTests.cs

```diff
@@ -2449,11 +2841,35 @@ public enum TestEnum
             Case3
         }
 
+        private static G[] CreateStructArray()
+        {
+            return new G[]
+            {
+                new G { x = 1, s = "Hello1", z = 2 },
+                new G { x = 2, s = "Hello2", z = 3 },
+                new G { x = 3, s = "Hello3", z = 4 },
+                new G { x = 4, s = "Hello4", z = 5 },
+                new G { x = 5, s = "Hello5", z = 6 }
+            };
+        }
+
         private struct G
```

Comment:
Nit: I know you're not adding this in this PR, but as an additional commit, could you rename this?  "G" isn't very descriptive :) and it makes it more difficult to understand the added test cases.